tmt/base.py

```diff
@@ -846,10 +846,10 @@ def _expand_node_data(self, data):
             data_expanded_vars = os.path.expandvars(data)
             # convert context values to string and define env variables using context
             # keys and values
-            list_context_string = [
-                (key, ','.join(value)) for (key, value) in self._fmf_context().items()
-                ]
-            with tmt.utils.modify_environ(list_context_string):
+            dict_context_vars = {
+                key: ','.join(value) for (key, value) in self._fmf_context().items()
```

Comment:
Last thing I am not sure about is if we should join list items as comma separated or space separated.